pkg/operator/workload/sync_openshift_oauth_apiserver.go

```diff
@@ -79,16 +78,21 @@ func NewOAuthAPIServerWorkload(
 
 // PreconditionFulfilled is a function that indicates whether all prerequisites are met and we can Sync.
 func (c *OAuthAPIServerWorkload) PreconditionFulfilled(ctx context.Context) (bool, error) {
-	authOperator, err := c.operatorClient.Authentications().Get(ctx, "cluster", metav1.GetOptions{})
+	operatorSpec, operatorStatus, _, err := c.operatorClient.GetOperatorState()
 	if err != nil {
 		return false, err
 	}
 
-	return c.preconditionFulfilledInternal(authOperator)
+	return c.preconditionFulfilledInternal(operatorSpec, operatorStatus)
 }
 
-func (c *OAuthAPIServerWorkload) preconditionFulfilledInternal(authOperator *operatorv1.Authentication) (bool, error) {
-	argsRaw, err := GetAPIServerArgumentsRaw(authOperator.Spec.OperatorSpec)
+func (c *OAuthAPIServerWorkload) preconditionFulfilledInternal(operatorSpec *operatorv1.OperatorSpec, operatorStatus *operatorv1.OperatorStatus) (bool, error) {
+	if operatorStatus.LatestAvailableRevision == 0 {
+		// this a backstop during the migration from 4.17 whe this information is in .status.oauthAPIServer.latestAvailableRevision
+		return false, fmt.Errorf("waiting for .status.latestAvailableRevision to be available")
+	}
```

Comment:
nice!

Reply:
@deads2k QE found a Blocker bug OCPBUGS-42932 related  to this.